recipes/graphvite/patches/0003-point-to-correct-faiss-lib.patch

```diff
@@ -0,0 +1,89 @@
+From 43268138b14ffb5c4912b392171d21c08fcd35ce Mon Sep 17 00:00:00 2001
+From: "H. Vetinari" <h.vetinari@gmx.com>
+Date: Sat, 12 Feb 2022 14:11:30 +1100
+Subject: [PATCH 3/4] point to correct faiss lib
+
+---
+ CMakeLists.txt               | 10 +++-------
+ python/graphvite/__init__.py |  4 +---
+ python/setup.py              |  6 +-----
+ src/CMakeLists.txt           |  2 +-
+ 4 files changed, 6 insertions(+), 16 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 28f8b7c..6bc112d 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -107,10 +107,9 @@ if (NOT FAISS_PATH)
+         set(FAISS_LIBRARY ${PROJECT_BINARY_DIR}/faiss/lib/libfaiss.so)
+     endif()
+ else()
+-    get_filename_component(FAISS_PARENT ${FAISS_PATH} DIRECTORY)
+-    include_directories(${FAISS_PARENT})
+-    link_directories(${FAISS_PATH})
+-    set(FAISS_LIBRARY ${FAISS_PATH}/libfaiss.so)
++    include_directories(${PROJECT_BINARY_DIR}/faiss/include)
++    link_directories(${PROJECT_BINARY_DIR}/lib)
++    set(FAISS_LIBRARY ${PROJECT_BINARY_DIR}/lib/libfaiss.so)
+ endif()
+ 
+ if (NOT CMAKE_BUILD_TYPE)
+@@ -138,9 +137,6 @@ endforeach()
+ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+-# symbolic link to faiss
+-file(MAKE_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
+-execute_process(COMMAND ln -sf ${FAISS_LIBRARY} ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libfaiss.so)
+ 
+ # project directories
+ include_directories(${PYTHON_INCLUDE_DIRS})
+diff --git a/python/graphvite/__init__.py b/python/graphvite/__init__.py
+index 876fe0e..7e2fb5f 100644
+--- a/python/graphvite/__init__.py
++++ b/python/graphvite/__init__.py
+@@ -26,9 +26,7 @@ from . import util
+ 
+ package_path = os.path.dirname(__file__)
+ candidate_paths = [
+-    os.path.realpath(os.path.join(package_path, "lib")),
+-    os.path.realpath(os.path.join(package_path, "../../lib")),
+-    os.path.realpath(os.path.join(package_path, "../../build/lib"))
++    os.path.realpath(os.path.join(os.environ["PREFIX"], "lib")),
```

Comment:
```suggestion
+    os.path.realpath(os.path.join(os.environ["CONDA_PREFIX"], "lib")),
```
`PREFIX` is unique to the *build* environment; it is where things should be installed. However, it will not be set in the user's runtime environment, so if you need to find this directory at runtime, then you should search `CONDA_PREFIX/lib`.

Reply:
Good catch, thanks.

Reply:
Seems the tests don't like that change though...

Reply:
Is the lib going to `PREFIX` or `SP`?

Reply:
also, this will still happen during the build, no? You trigger setup during the build anyway unless the library is built elsewhere, which doesn't to be the case here --- or is it? I would think `PREFIX` is the correct one...

Reply:
Generally, if a shared library is intended to only be consumed as a python module (e.g. only the pybind11 python API is considered public), then it should be shipped inside site-packages and treated as any other python module. e.g. the files `graphvite/foo.py` and `graphvite/bar.so` would be imported like `import graphvite.foo` `import graphvite.bar`. A shared library with only a C interface that is linked to another compiled file should go in `PREFIX/libs`. I think you have both types of files, but I didn't look to closely. Feel free to reach out if you have questions about this, but I think your recipe will ship a working package.

Reply:
Fair point, thanks for spelling that out so coherently. To be honest, I also haven't verified the interface (python vs. C), but my default was to put the `.so` into `$PREFIX/lib`. In any case, I'm 99% sure this will work. I would have liked to run more tests in CI, but couldn't find a test suite (at first glance) upstream.